utils.ts

```diff
@@ -9,26 +9,60 @@ import {join} from 'path';
 import {execSync} from 'child_process';
 import * as mkdirp from 'mkdirp';
 
-export function defaults<T>(val: T | undefined, def: T): T {
+export function defaults<T>(val: T|undefined, def: T): T {
   return typeof val === 'undefined' ? def : val;
 }
 
 export async function delay(time: number) {
   return new Promise(resolve => setTimeout(resolve, time));
 }
 
+export function getPlatform() {
+  const platform = process.platform;
+
+  if (platform !== 'linux') {
+    return platform;
+  }
+  // Windows Subsystem for Linux, reports Linux as platform but chrome is on Windows
+  // we need to find out the real platform
+  if (process.env.PATH && (process.env.PATH as string).includes(':/mnt/c/Windows/System32:')) {
+    return 'wsl';
+  }
+
+  return platform;
+}
+
 export function makeTmpDir() {
-  switch (process.platform) {
+  switch (getPlatform()) {
     case 'darwin':
     case 'linux':
       return makeUnixTmpDir();
+    case 'wsl':
+      // We populate the user's Windows temp dir so the folder is correctly created later
+      process.env.TEMP = `/mnt/c/Users/${findWSLUser()}/AppData/Local/Temp`;
     case 'win32':
       return makeWin32TmpDir();
     default:
       throw new Error(`Platform ${process.platform} is not supported`);
   }
 }
 
+export function toWinDirFormatIfNeeded(dir: string|any): string|any {
+  if (!dir || getPlatform() !== 'wsl') {
+    return dir;
+  }
+
+  return dir.replace('/mnt/c/', 'C:\\').replace(/\//g, '\\');
+}
+
+export function findWSLUser(): string {
```

Comment:
Everywhere this is used, it's mostly just to resolve the LOCALAPPDATA path. how about we change this method to be a `getLocalAppDataPath()`?

Reply:
Done